metadata-ingestion/tests/unit/sdk_v2/test_lineage_client.py

```diff
@@ -379,3 +384,148 @@ def test_add_dataset_transform_lineage_complete(client: DataHubClient) -> None:
         column_lineage=column_lineage,
     )
     assert_client_golden(client, _GOLDEN_DIR / "test_lineage_complete_golden.json")
+
+
+def test_add_dataset_lineage_from_sql(client: DataHubClient) -> None:
+    """Test adding lineage from SQL parsing with a golden file."""
+
+    lineage_client = LineageClient(client=client)
+
+    # Create upstream and downstream datasets
+    client.entities.upsert(
+        Dataset(
+            platform="snowflake",
+            name="orders",
+            env="PROD",
+            schema=[("price", "float"), ("qty", "int"), ("unit_cost", "float")],
+        )
+    )
+    client.entities.upsert(
+        Dataset(platform="snowflake", name="sales_summary", env="PROD")
+    )
+
+    # Create minimal mock result with necessary info
+    mock_result = SqlParsingResult(
+        in_tables=["urn:li:dataset:(urn:li:dataPlatform:snowflake,orders,PROD)"],
+        out_tables=[
+            "urn:li:dataset:(urn:li:dataPlatform:snowflake,sales_summary,PROD)"
+        ],
+        column_lineage=[],  # Simplified - we only care about table-level lineage for this test
+        query_type=QueryType.SELECT,
+        debug_info=MagicMock(error=None, table_error=None),
+    )
+
+    # Simple SQL that would produce the expected lineage
+    query_text = "SELECT price, qty, unit_cost FROM orders"
```

Comment:
we can just use a query that creates output tables

e.g. `create table sales_summary as select ...` 

also, this should test a query that has multiple input tables and has CLL

Reply:
done